### PR TITLE
Add support for augeas loading of non stock lenses

### LINKED ISSUE #30131: Augeas modules should support loading of non stock lenses
It would be great to have augeas support non stock lenses by allowing the user to specify a list of additional directories which should be searched for lenses.

### PR DESCRIPTION
**Pull request by akissa**
Fixes #30131

Pylint cleanup

Updated tests

### CODE REVIEW

File: salt/modules/augeas_cfg.py [hunk 1]
```diff
@@ -100,7 +100,29 @@ def _lstrip_word(word, prefix):
     return word
 
 
-def execute(context=None, lens=None, commands=()):
+def _check_load_paths(load_path):
+    '''
+    Checks the validity of the load_path, returns a sanitized version
+    with invalid paths removed.
+    '''
+    if load_path is None or not isinstance(load_path, six.string_types):
```
Comment (cachedout) [inline salt/modules/augeas_cfg.py]: Isn't this redundant?

**Comment by akissa** [comment]
It is just a safe guard to stop someone being able to send in a list for example in a future update to the state module for example.

**Comment by cachedout** [comment]
Gotcha. I figured you had a reason. I was mostly just curious. :]

Thanks, @akissa

**Comment by akissa** [comment]
@cachedout Happy new year and Thanks

**Comment by cachedout** [comment]
Cheers! You as well, @akissa

**Pull request merged**